add-chain/contract_addresses.go

```diff
@@ -0,0 +1,160 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/ethereum-optimism/superchain-registry/superchain"
+)
+
+type AddressData struct {
+	Address string `json:"address"`
+}
+
+var (
+	// Addresses to retrieve from JSON
```

Comment:
Can we add the DA Challenge contract? It's also possible to merge this first before adding support for plasma mode.

Reply:
Created issue [here](https://github.com/ethereum-optimism/superchain-registry/issues/221) to capture that work